files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/appGroup.js

```diff
@@ -103,6 +102,10 @@ class AppGroup {
         });
         this.actor._delegate = this;
 
+        this.spacing = 0;
+        this.actor.connect('style-changed',
+            () => this.spacing = this.actor.peek_theme_node().get_length('spacing'));
```

Comment:
I think this is negatively affecting performance. I added a log here, and this signal fires on every allocation change. Input latency improves when I move this line:

```js
this.spacing = this.actor.peek_theme_node().get_length('spacing')
```

to `onIconBoxStyleChanged` - we already have `iconBox` connecting to `style-changed`, and its one too many IMHO - we have three other allocation signals, and they are firing at the same frequency.

We don't need to worry about theme changing because the `theme-set` signal callback does a destroy/create cycle.  You could add this to _init, and it should take care of it.

```js
setTimeout(() => this.actor.style_length('spacing'), 0);
```

`get_length` is a heavy call that starts a cascade of C function calls, and we shouldn't call it on every allocation. I've been trying to optimize their usage to mixed results. St.Widget.prototype.style_length from environment.js guards against finalized/unrealized actors, and in #8092 it caches the theme node. I think the best way forward is to only use theme introspection we need - #8091 and #8084 demonstrates none of this is necessary, at least right now.

I can't see a difference when I comment out spacing entirely - I'm looking for the things that I fixed in the PRs, and I'm not sure its worth the additional complexity here.